internal/commands/list.go

```diff
@@ -94,38 +123,131 @@ func printHeaders(w io.Writer) {
 	fmt.Fprintln(w, strings.Join(headers, "\t"))
 }
 
-func printValues(w io.Writer, installation *store.Installation) {
+func printValues(w io.Writer, installation Installation) {
 	var values []string
 	for _, column := range listColumns {
 		values = append(values, column.value(installation))
 	}
 	fmt.Fprintln(w, strings.Join(values, "\t"))
 }
 
-func getInstallations(targetContext, configDir string) ([]*store.Installation, error) {
-	appstore, err := store.NewApplicationStore(configDir)
-	if err != nil {
-		return nil, err
-	}
-	installationStore, err := appstore.InstallationStore(targetContext)
-	if err != nil {
-		return nil, err
-	}
+type Installation struct {
+	*store.Installation
+	Services appServices `json:",omitempty"`
+}
+
+func getInstallations(installationStore store.InstallationStore, fetcher ServiceFetcher) ([]Installation, error) {
 	installationNames, err := installationStore.List()
 	if err != nil {
 		return nil, err
 	}
-	installations := make([]*store.Installation, len(installationNames))
+	installations := make([]Installation, len(installationNames))
+	var errs []string
 	for i, name := range installationNames {
 		installation, err := installationStore.Read(name)
 		if err != nil {
 			return nil, err
 		}
-		installations[i] = installation
+		services, err := fetcher.getServices(installation)
+		if err != nil {
+			errs = append(errs, err.Error())
+		}
+		installations[i] = Installation{Installation: installation, Services: services}
 	}
 	// Sort installations with last modified first
 	sort.Slice(installations, func(i, j int) bool {
 		return installations[i].Modified.After(installations[j].Modified)
 	})
-	return installations, nil
+	if len(errs) > 0 {
+		err = errors.New(strings.Join(errs, "\n"))
+	}
+	return installations, err
+}
+
+type ServiceStatus struct {
+	DesiredTasks int
+	RunningTasks int
+}
+
+type appServices map[string]ServiceStatus
+
+type runningService struct {
+	Spec struct {
+		Name string
+	}
+	ServiceStatus ServiceStatus
+}
+
+type serviceFetcher struct {
```

Comment:
Shouldn't it be better in a dedicated package outside `commands`?

Reply:
Ok I'll try to extract that part and see if it's better 👍

Reply:
Tried but use too many things in `command` package I'd have to extract too and refactor. Should be done in a followup.